### ISSUE TITLE
About negative learning

### ISSUE THREAD

**Issue by chengjianhong** [nayeemrizve/ups#5]
Hi, thanks for your this excellent work. I understand negative learning as assigning a new category if certain conditions are met. I can't understand the following code. What is the effect of dividing the output by the temp_nl?  

`out_prob_nl.append(F.softmax(outputs/args.temp_nl, dim=1)) #for selecting negative pseudo-labels`

**Comment by nayeemrizve** [comment]
Hi, thank you very much for expressing your interest in our work. 

We have mentioned the reasoning behind temperature scaling for negative learning in the implementation details. I am quoting the text from the paper in the following: 

> To further reduce the calibration error and to make the negative pseudo-label selection more robust, we perform temperature scaling to soften the output predictions - we set T = 2 (Guo et al., 2017).

You can also go through the "Temperature scaling" paragraph of section 4.2 of _On Calibration of Modern Neural Networks_ paper for more details.

**Comment by chengjianhong** [comment]
If my task is just a binary classification task,  can you give me some advice?
